### PR TITLE
Bugfix format for bootstrap configuration file.

### PR DESCRIPTION
**Pull request by wangfakang**
The JSON spec doesn't allow trailing comma in an array or object. However, in some scenarios, the `pilot-agent` will render some illegal JSON via `envoy_bootstrap_v2.json` template file. 
   
 ```  
   "stats_matcher": {
      "inclusion_list": {
        "patterns": [
          {
          "prefix": "reporter="
          },
          {
          "prefix": "component"
          },
          {{- range $a, $s := .inclusionPrefix }}
          {
          "prefix": "{{$s}}"
          },
          {{- end }}
          {{- range $a, $s := .inclusionSuffix }}
          {
          "suffix": "{{$s}}"
          },
          {{- end }}
          {{- range $a, $s := .inclusionRegexps }}
          {
          "regex": "{{js $s}}"
          },
          {{- end }}
        ]
      }
    }
```

maybe render results in a scene without `inclusionRegexps` are as follows:
	
```
    "stats_matcher": {
      "inclusion_list": {
        "patterns": [
          {
          "prefix": "reporter="
          },
          {
          "prefix": "component"
          },
          {
          "prefix": "cluster_manager"
          },
          {
          "prefix": "listener_manager"
          },
          {
          "prefix": "http_mixer_filter"
          },
          {
          "prefix": "tcp_mixer_filter"
          },
          {
          "prefix": "server"
          },
          {
          "prefix": "cluster.xds-grpc"
          },
          {
          "suffix": "ssl_context_update_by_sds"
          },
        ]
      }
    }
```

### CODE REVIEW

**Comment by howardjohn** [comment]
You should run `REFRESH_GOLDEN=true go test ./pkg/bootstrap/...` to update the test files

**Comment by howardjohn** [comment]
Just wondering, what is  the motivation? Envoy is fine with trailing spaces from my testing.

I am not against making it valid json, just curious

**Comment by wangfakang** [comment]
> Just wondering, what is the motivation? Envoy is fine with trailing spaces from my testing.
> 
> I am not against making it valid json, just curious

We are adapting [MOSN](https://github.com/mosn/mosn) to the standard Istio, and  we hope that the MOSN can be one of the cloud native sidecars, Here's more on that [mosn-isto-wg](https://github.com/mosn/community/blob/master/wg-istio.md).

**Comment by wangfakang** [comment]
@howardjohn In the field of Service Mesh, Istio is already a standard in the control plane. So we also hope that MOSN can be one of the standard data palne in istio. The advantage of this is that users can make flexible choices on the data plane according to their own scenarios.I am looking forward to your Suggestions, thank you.

**Comment by wangfakang** [comment]
> You should run `REFRESH_GOLDEN=true go test ./pkg/bootstrap/...` to update the test files

@howardjohn It has been updated, thx.

File: tools/packaging/common/envoy_bootstrap_v2.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
       "zone": "{{ .zone }}",
       {{- end }}
       {{- if .sub_zone }}
-      "sub_zone": "{{ .sub_zone }}",
```
Comment (howardjohn) [inline tools/packaging/common/envoy_bootstrap_v2.json:13]: if you don't have a subzone you will still get a trailing `,`

Comment (wangfakang) [reply tools/packaging/common/envoy_bootstrap_v2.json:13]: thx,fixed it.

File: pkg/bootstrap/testdata/runningsds_golden.json [hunk 2]
```diff
@@ -3,9 +3,11 @@
     "id": "sidecar~1.2.3.4~foo~bar",
     "cluster": "istio-proxy",
     "locality": {
-      "region": "regionA",
-      "zone": "zoneB",
-      "sub_zone": "sub_zoneC",
+      "region": "regionA"
+      ,
```
Comment (howardjohn) [inline pkg/bootstrap/testdata/runningsds_golden.json:7]: is it legal to have comma on new line?

Comment (wangfakang) [reply pkg/bootstrap/testdata/runningsds_golden.json:7]: This is consistent with the JSON specification，we can use  [json tools](https://jsonlint.com/) to validate it.
image

**Comment by wangfakang** [comment]
@howardjohn  Now a test case is [faild](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/22994/integ-telemetry-k8s-tests_istio/11395), seems `FAIL	istio.io/istio/tests/integration/telemetry/stats/prometheus/tcp	348.476s` , But in my environment it is passed,  such as:
```
master $ go test ./tests/integration/telemetry/stats/prometheus/tcp/
ok      istio.io/istio/tests/integration/telemetry/stats/prometheus/tcp (cached)
```

**Comment by howardjohn** [comment]
/retest

looks like maybe a flake

**Pull request merged**